### PR TITLE
Migrate pytorch-linux-bionic-py3.8-gcc9-coverage to GHA

### LINKED ISSUE #59166: Running some CI tests with `xlarge` `resource_class` VMs for testing AVX512 support
## 🚀 Feature

Currently, the Linux CI checks that are tests are run on `large` `resource_class` executors, which are run on machines that don't support AVX512. To test ATen AVX512 kernels, `xlarge` executors would be required, as these VMs are run on Intel Cascade Lake machines that support AVX512.

## Pitch

Some CI checks should be run on `xlarge` instances, which have AVX512 support. The number of CI tests running on `xlarge` instances can probably be limited in order to not vastly increase CI costs.

cc @ezyang @seemethere @malfet @walterddr @pytorch/pytorch-dev-infra @ngimel @VitalyFedyunin

### PR DESCRIPTION
**Pull request by imaginary-person**
Fixes #59166

`pytorch-linux-bionic-py3.8-gcc9-coverage` build & tests can be run on `linux.2xlarge` instances on GHA,
which have AVX512 support.

Thanks

cc @malfet @seemethere @samestep @zhouzhuojie

### CODE REVIEW

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
 
-GITHUB_DIR = Path(__file__).parent.parent
+GITHUB_DIR = Path(__file__).resolve().parent.parent
```
Comment (imaginary-person) [inline .github/scripts/generate_ci_workflows.py:11]: `resolve` is required on Ubuntu (and maybe some other Linux distros).

File: .circleci/cimodel/data/pytorch_build_data.py [hunk 2]
```diff
@@ -80,17 +80,6 @@
...
-            ("9", [
-                ("3.8", [
-                    ("coverage", [
-                        (True, [
-                            ("shard_test", [XImportant(True)]),
-                        ]),
-                    ]),
-                ]),
-            ]),
-        ]),
```
Comment (imaginary-person) [inline .circleci/cimodel/data/pytorch_build_data.py:93]: These lines were manually removed & then `regenerate.sh` was run to create `config.yml`.

**Review by samestep — APPROVED** [review]
lgtm, thank you! @pytorch/pytorch-dev-infra just to check, are we cool with enabling this in GHA and disabling it in CircleCI in the same PR?

**Comment by walterddr** [comment]
Is there a before/after codecov result comparison done for this PR?
I've been looking into the codecov drop between https://github.com/pytorch/pytorch/compare/5bc28c8...03b5a22 and it seems like this one might've been related. is there a codecov report generated for this PR that I can download and take a look?

**Pull request closed without merging**